### PR TITLE
[Feature] Simple protocol support

### PR DESCRIPTION
**Pull request by kodomir**

### CODE REVIEW

File: BB/Arguments/Argue.swift [hunk 1]
```diff
@@ -15,7 +15,7 @@ extension String {
             let startIndex = self.startIndex.advancedBy(r.startIndex)
             let endIndex = startIndex.advancedBy(r.endIndex - r.startIndex)
 
-            return self[Range(start: startIndex, end: endIndex)]
+            return self[Range(startIndex..<endIndex)]
```
Comment (insanehunter) [inline BB/Arguments/Argue.swift:18]: Note to self: check if Range() constructor is really necessary.

**Comment by insanehunter** [comment]
Awesome piece of work, thanks.

I'be got a question though - what does it give aside from having a protocol declared in the same spec file as everything else? I mean, protocol is pretty much a declaration, no boilerplate code generation is needed, why not declare it in a separate file?

**Comment by kodomir** [comment]
If you are using code generator for network classes, it's really nice to keep all structs and etc together. It's just could help you don't miss something and check all properties before compilation.

**Comment by insanehunter** [comment]
Could you please provide an example?

**Comment by insanehunter** [comment]
Oh, I finally got it – it's useful for generating protocol extensions. Nice.

**Pull request merged**